app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt

```diff
@@ -185,6 +186,13 @@ class SearchRobot {
             mDevice.waitForIdle()
             closeSoftKeyboard()
             mDevice.pressBack()
+            try {
+                assertTrue(searchWrapper().waitUntilGone(waitingTimeShort))
+            } catch (e: AssertionError) {
+                mDevice.pressBack()
+                assertTrue(searchWrapper().waitUntilGone(waitingTimeShort))
+            }
+
```

Comment:
I think we should just wrap everything, including 186-188 into a Try-Catch-Finally.